thrift/bitset.go

```diff
@@ -0,0 +1,70 @@
+package thrift
+
+import "errors"
+
+const size = 64
+
+type bits uint64
+
+// BitSet is a set of bits that can be set, cleared and queried.
+type BitSet []bits
+
+func numChange(t interface{}) (uint64, error) {
+
+	switch t := t.(type) {
+	case int:
+		return uint64(t), nil
+	case uint:
+		return uint64(t), nil
+	case int8:
+		return uint64(t), nil
+	case int16:
+		return uint64(t), nil
+	case int32:
+		return uint64(t), nil
+	case int64:
+		return uint64(t), nil
+	case uint8:
+		return uint64(t), nil
+	case uint16:
+		return uint64(t), nil
+	case uint32:
+		return uint64(t), nil
+	case uint64:
+		return uint64(t), nil
+	}
+	return 0, errors.New("type error")
+}
+
+// Set ensures that the given bit is set in the BitSet.
+func (s *BitSet) Set(n interface{}) {
```

Comment:
Should change these to just take `int` as it's only used internally and the type can be guaranteed. Will be slightly more efficient.